src/js/Alert.js

```diff
@@ -1,18 +1,77 @@
 import React from 'react';
 
-import {AlertStrings} from './containers/Metadata'
-import {AlertButtons} from './containers/Buttons';
+import { AlertStrings } from './containers/Metadata'
+import { AlertButtons } from './containers/Buttons';
 import Image from './Templates/Shared/Image'
 import StaticIcon from './Templates/Shared/StaticIcon'
+import store from './store.js'
+import { alertTimeoutReseted } from './actions'
+import UIController from './Controllers/UIController'
+import TTSController from './Controllers/TTSController'
+import EventEmitter from "reactjs-eventemitter";
 
+const OUT_OF_BOUND_RESET_PERIOD = 1000;
 export default class Alert extends React.Component {
+    constructor(props) {
+        super(props);
+        this.state = {
+            alertCounter: 5,
+            ifSpeak: true,
+            speakCounter: 5,
+            ttsStoped: false
+        }
+        this.pressResetTimeoutButton = this.pressResetTimeoutButton.bind(this);
+        EventEmitter.subscribe('TTSTimeout', () => {
+            this.setState({ ttsStoped: true });
+        })
+    }
+
+    pressResetTimeoutButton(event) {
+        store.dispatch(alertTimeoutReseted(true));
+
+        let count = store.getState().ui[store.getState().activeApp].resetTimeout.resetTimeoutValue / 1000;
+        if (count > OUT_OF_BOUND_RESET_PERIOD) {
+            EventEmitter.emit('OUT_OF_BOUND');
+            return;
+        }
+        this.setState({ alertCounter: count });
+        UIController.resetAlertTimeout()
+
+    }
+    changeCounter() {
+        this.setState(prevState => ({ alertCounter: prevState.alertCounter > 0 ? prevState.alertCounter - 1 : '' }))
+        this.setState(prevState => ({ speakCounter: prevState.speakCounter > 0 ? prevState.speakCounter - 1 : '' }))
+    }
+    componentDidMount() {
+        this.setState({alertCounter: store.getState().ui[store.getState().activeApp].alert.duration/1000});
+        this.interval = setInterval(() => this.changeCounter(), 1000);
+    }
+    componentWillUnmount() {
+        clearInterval(this.interval)
+    }
+
+
     render() {
         var fill = this.props.theme ? "#FFFFFF" : "#000000";
         var icon = this.props.icon ? this.props.icon : { imageType: "STATIC", value: "0xFE" }
 
         var iconElement = (icon.imageType === "STATIC")
-                 ? (<StaticIcon class="alert-icon" image={icon.value} />)
-                 : (<div className="alert-icon"><Image class="icon" image={icon.value} isTemplate={icon.isTemplate} fillColor={fill} /></div>);
+            ? (<StaticIcon class="alert-icon" image={icon.value} />)
+            : (<div className="alert-icon"><Image class="icon" image={icon.value} isTemplate={icon.isTemplate} fillColor={fill} /></div>);
+
+        let resetTimeoutHTML = undefined;
```

Comment:
@iCollin If we remove the button, we cant to check reset timeout for Alert. Are you OK with it ?

Reply:
Do you mean you can't see how much time remains on the Alert? We've never had that before, I don't think that is necessary. Otherwise I am not sure what you mean.

Reply:
@iCollin, Valerii said about checking reset timeout for core in this case.
According to current SDL logic, SDL does not start timeout in case it receives `Alert` request with `softButtons` and provides `duration=0` to the HMI (you can find info about duration here https://github.com/smartdevicelink/sdl_hmi_integration_guidelines/blob/b06288fc58090b28ba4e8aed7119292388798eca/docs/UI/Alert/index.md )
Because of this SDL logic, HMI should wait for the user action on Alert request with `softButtons` and SDL will wait as much time as the user needs.

SDL can not reset its timeout by receiving `OnResetPeriod` notification for the Alert request with `softButtons` from HMI  because it has no timeout for this request.

SDL sets timeout only for Alert request without `softButtons`, so the case with actual resetting timeout inside SDL can not be checked in case of removing additional button for reset.

In case you are okay with such HMI behavior(sending OnResetTimeout by KEEP_CONTEX button press only) according to the information above we can remove all additional buttons.

Note: Such SDL logic is applicable for Alert and SubtleAlert RPCs.

Reply:
@iCollin, We propose to add logic for Alert RPC without `softButtons` to send OnResetTimeoutnotification for `UI.Alert` request in case speak needs more time than defined in the request (for now HMI resets timeout only for TTS.Speak request in the current case)

Reply:
@GetmanetsIrina thank you for explaining.

I see what you're saying. I think what you propose is okay, but only if the mobile app does not specify a duration. If the mobile sends a request with specific duration, that should not be modified or reset.

Reply:
@iCollin, but HMI will always receive the duration in `UI.Alert` request, because there is the default value in case a mobile request is received without duration.

Reply:
Thank you for explaining, yes I agree what you proposed will be necessary.

Reply:
@iCollin There was fixed response for TTS.Speak in commit c75780bfd1e582ea71c4b8b3c72f99ea8ae1a463
Additional logic to check if TTS.Speak in progress while alert timeout expired will be ready tommorow

Reply:
@iCollin Added logic for reset timeout if TTS is in progress d078b47e795c3d81bbf1b56a681842ad7c63e738